### ISSUE TITLE
feat: Create config.json if not exists

### ISSUE THREAD

**Issue by gaocegege** [litanlitudan/skyagi#82]
Or we will get this:

```
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ /home/gaocegege/code/skyagi/skyagi/src/skyagi/cli.py:264 in main                                 │
│                                                                                                  │
│   261 │   if ctx.invoked_subcommand is not None:                                                 │
│   262 │   │   return                                                                             │
│   263 │                                                                                          │
│ ❱ 264 │   run()                                                                                  │
│   265                                                                                            │
│                                                                                                  │
│ ╭────────────────────── locals ───────────────────────╮                                          │
│ │ ctx = <click.core.Context object at 0x7f6febe99520> │                                          │
│ ╰─────────────────────────────────────────────────────╯                                          │
│                                                                                                  │
│ /home/gaocegege/code/skyagi/skyagi/src/skyagi/cli.py:167 in run                                  │
│                                                                                                  │
│   164 │   """                                                                                    │
│   165 │   Run SkyAGI                                                                             │
│   166 │   """                                                                                    │
│ ❱ 167 │   settings = Settings()                                                                  │
│   168 │                                                                                          │
│   169 │   # Ask Model settings                                                                   │
│   170 │   questions = [                                                                          │
│                                                                                                  │
│ /home/gaocegege/code/skyagi/pydantic/env_settings.py:40 in                                       │
│ pydantic.env_settings.BaseSettings.__init__                                                      │
│                                                                                                  │
│ [Errno 2] No such file or directory: '/home/gaocegege/code/skyagi/pydantic/env_settings.py'      │
│                                                                                                  │
│ /home/gaocegege/code/skyagi/pydantic/env_settings.py:75 in                                       │
│ pydantic.env_settings.BaseSettings._build_values                                                 │
│                                                                                                  │
│ [Errno 2] No such file or directory: '/home/gaocegege/code/skyagi/pydantic/env_settings.py'      │
│                                                                                                  │
│ /home/gaocegege/code/skyagi/pydantic/utils.py:216 in pydantic.utils.deep_update                  │
│                                                                                                  │
│ [Errno 2] No such file or directory: '/home/gaocegege/code/skyagi/pydantic/utils.py'             │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
AttributeError: 'str' object has no attribute 'items'
```

**Comment by qizheng7** [comment]
@gaocegege 
hmm.....I couldn't reproduce this issue. This is what I saw if config.json doesn't exist:

```
(venv-skyai-release) qiz@docker-desktop:~/skyagi$ skyagi
Please config your OpenAI token before using this app
Config by running: skyagi config openai or OPENAI_API_KEY="..." skyagi
You didn't provide an API key. You need to provide your API key in an Authorization header using Bearer auth (i.e. Authorization: Bearer YOUR_KEY), or as the password field (with blank username) if you're accessing the API from your browser and are prompted for a username and password. You can obtain an API key from https://platform.openai.com/account/api-keys.
(venv-skyai-release) qiz@docker-desktop:~/skyagi$ ls ~/.skyagi/
(venv-skyai-release) qiz@docker-desktop:~/skyagi$ rmdir ~/.skyagi
(venv-skyai-release) qiz@docker-desktop:~/skyagi$ skyagi
Please config your OpenAI token before using this app
Config by running: skyagi config openai or OPENAI_API_KEY="..." skyagi
You didn't provide an API key. You need to provide your API key in an Authorization header using Bearer auth (i.e. Authorization: Bearer YOUR_KEY), or as the password field (with blank username) if you're accessing the API from your browser and are prompted for a username and password. You can obtain an API key from https://platform.openai.com/account/api-keys.
```

**Comment by qizheng7** [comment]
and my skyagi release version is
```
(venv-skyai-release) qiz@docker-desktop:~/skyagi$ pip show skyagi
Name: skyagi
Version: 0.0.28
```

**Comment by gaocegege** [comment]
```
Name: skyagi
Version: 0.0.29.dev5+g2b82b0c.d20230523
Summary: AGI
Home-page: https://github.com/litanlitudan/skyagi
Author: Tan Li
Author-email: tan@tanli.dev
License: 
Location: /home/gaocegege/applications/miniconda3/envs/dev/lib/python3.9/site-packages
Requires: discord.py, faiss-cpu, inquirer, langchain, openai, pinecone-client, rich, termcolor, textual, tiktoken, typer
Required-by: 
```

I install it with make install.

**Comment by qizheng7** [comment]
got it. i can reproduce it locally. will fix it.

**Comment by gaocegege** [comment]
Thanks! By the way, it would be better to keep [XDG spec](https://wiki.archlinux.org/title/XDG_Base_Directory) for the config.json:

Currently: `$HOME/.skyagi/config,json`

XDG: `$HOME/.config/skyagi/config,json`

XDG could support macOS, windows and some other os. Maybe we could use the package [`xdg`](https://pypi.org/project/xdg/)

**Comment by headwinds** [comment]
<img width="770" alt="Screen Shot 2023-05-23 at 10 29 30 AM" src="https://github.com/litanlitudan/skyagi/assets/68402/718be8a5-35c9-47eb-8adc-5c9a2a4c1326">

I'm also getting a similar error. What if you included a config-example.json that we can simply copy and rename to config.json and supply our own open ai token? 

Exciting project, and am learning lots from the code base itself!

**Comment by headwinds** [comment]
<img width="1015" alt="Screen Shot 2023-05-23 at 11 00 08 AM" src="https://github.com/litanlitudan/skyagi/assets/68402/44c7bfee-6bbe-44a9-9392-8e03d458b571">

ha... I got it to start by hardcoding to supply the settings config 

```
 settings = {"model":"text-davinci-003"}
 ctx = agi_init(agent_configs, console, settings, user_index)
```    

But I don't know all the key value pairs... and it got stuck when on the next time it needed a key/value for `embedding`

```
embeddings_model = load_embedding_from_config(ctx.settings.model.embedding) 
``` 

Anyways...obviously a hack... would be much better to see & generate the proper config.json

**Comment by qizheng7** [comment]
thanks @headwinds  for sharing the info. we are working on a proper fix now. will push it out soon.

**Comment by qizheng7** [comment]
@gaocegege @headwinds the fix is in PR #84

**Comment by gaocegege** [comment]
@qizheng7 

Thanks!

**Comment by headwinds** [comment]
@qizheng7 `skyagi config --help` is a sweet command - you should consider adding it to the README.md - feels important for new devs

I tried again using that command and its working - super cool!

![Screen Shot 2023-05-24 at 9 13 51 AM](https://github.com/litanlitudan/skyagi/assets/68402/8d15f219-a19f-4eaf-b6e7-d34de05d8d3f)

A CLI is certainly a nice way to test to an API and focus on the conversation.

**Comment by qizheng7** [comment]
@headwinds good suggestion! will do

**Comment by qizheng7** [comment]
close this issue. will resolve `XDG` spec and update README in a separate PR for CLI.
